File: network-client/src/main/com/mgtriffid/academ/network/client/NetworkClient.java
package com.mgtriffid.academ.network.client;

import com.esotericsoftware.kryonet.Client;
import com.mgtriffid.academ.network.common.PlayerCommand;

/**
 * Created by mgtriffid on 19.03.17.
 */
public class NetworkClient {

    private Client client;
    private CommandsChannel commandsChannel = new CommandsChannel();

    public void sendCommands(PlayerCommand command) {

    }

    public CommandsChannel provideCommandsChannel() {
        return commandsChannel;
    }

}

File: network-client/src/main/java/com/mgtriffid/academ/network/client/NetworkClient.kt
package com.mgtriffid.academ.network.client

import com.mgtriffid.academ.network.common.PlayerCommand
import com.mgtriffid.academ.network.common.commands.meta.EnterGameCommand

/**
 * Created by mgtriffid on 17.04.17.
 */
interface NetworkClient {
    fun send(enterGameCommand: EnterGameCommand)

    fun start()

    fun send(command: PlayerCommand)
}
